components/org.wso2.carbon.identity.branding.preference.resolver/src/main/java/org/wso2/carbon/identity/branding/preference/resolver/UIBrandingPreferenceResolverImpl.java

```diff
@@ -117,10 +132,23 @@ public BrandingPreference resolveBranding(String type, String name, String local
             }
         }
 
-        /* Tenant domain will always be carbon.super for SaaS apps (ex. myaccount). Hence need to resolve
+        if (APPLICATION_TYPE.equals(type)) {
+            return resolveApplicationBranding(name, locale, organizationManager, organizationId, currentTenantDomain);
+        } else if (ORGANIZATION_TYPE.equals(type)) {
+            return resolveOrganizationBranding(name, locale, organizationManager, organizationId, currentTenantDomain);
+        } else {
+            throw handleClientException(ERROR_CODE_INVALID_BRANDING_PREFERENCE_TYPE, type, getTenantDomain());
```

Comment:
getTenantDomain() can be replaced with currentTenantDomain

Reply:
Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/40/commits/05fa6da03cb482ecbd514d7030744011bad7a98e